### PR TITLE
Add analytics events

### PR DESCRIPTION
**Pull request by saperi22**
### Summary of changes

 - Add analytics events for Shopper Insights

 ### Checklist

 - [ ] Added a changelog entry

- @saperi22

### CODE REVIEW

File: ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt [hunk 1]
```diff
@@ -47,6 +51,7 @@ class ShopperInsightsClient @VisibleForTesting internal constructor(
                     )
                 )
             )
+            braintreeClient.sendAnalyticsEvent(GET_RECOMMENDED_PAYMENTS_FAILED)
```
Comment (MaxHastingsPP) [inline ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:54]: Would it be possible to put why it failed? Merchant passing bad data

File: ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt [hunk 2]
```diff
@@ -76,6 +82,7 @@ class ShopperInsightsClient @VisibleForTesting internal constructor(
                 )
             )
         )
+        braintreeClient.sendAnalyticsEvent(GET_RECOMMENDED_PAYMENTS_SUCCEEDED)
```
Comment (sarahkoop) [inline ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:85]: For other modules in v5, to make sure that we always send an analytics event with each callback, we've been creating helper methods like `callbackRecommendedPaymentsSuccess()` which would include the calling back and sending analytics (for success and failure).

**Review by sarahkoop — APPROVED** [review]
Take it or leave it comment on callback methods otherwise LGTM 👍

File: ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt (hunk 2 above)
Comment (saperi22) [reply ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:85]: Yep, I remember the pattern from one of your PRs. I did initially write it that way, but then I took a second look and realized I was abstracting out 2 calls into a method. I concluded that the abstraction wasn't worth it at this time.

File: ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt (hunk 1 above)
Comment (saperi22) [reply ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:54]: Max, I don't see us sending any data in any of the other places. We'll have to look into the API to figure out if it accepts a key, value data object. I'll keep this PR as is and figure that out as part of another ticket if the need arises.

Comment (MaxHastingsPP) [reply ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:54]: Makes sense.

**Review by MaxHastingsPP — APPROVED** [review]

**Pull request merged**